### DISCUSSION TITLE
How can I group my "lock-file maintenance" into fewer PRs?

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#28276] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I love the "lock-file maintenance" feature! However, I would like to group these into a single PR for a certain file pattern. I've tried this package rule but I can't find a way to test it locally:

```json
    {
      "description": "Backend service lock-file maintenance",
      "matchFileNames": ["services/*/pyproject.toml"],
      "matchUpdateTypes": ["lockFileMaintenance"],
      "additionalBranchPrefix": "backend-",
      "commitMessagePrefix": "backend:",
      "commitMessageAction": "Refresh lock-files"
    },
```

I usually test changes locally using something like this:

```shell
npx -y renovate \
  --dry-run=lookup \
  --token=REDACTED \
  --platform=local \
  --report-type=file \
  --report-path=.renovate-report.json \
  --force-cli=true \
  --include-paths='backend/services/**' \
;
```

This doesn't seem to work for lock-file maintenance updates 😕 

### Logs (if relevant)

(Let me know if this is useful for this issue.)

**Comment by secustor** [comment]
Lock file maintenance simply run the package managers update routine, therefore we can not simply split it. :/

**Comment by rarkins** [comment]
Do you have e a single PR today? And you want to split it?

**Reply by allanlewis** [reply]
No, today I have one PR per lock file and I want one combined PR for all lock files in a given pattern.

**Reply by rarkins** [reply]
By default there should be a single PR with branch name `renovate/lock-file-maintenance`. What branche names are you seeing?

**Reply by allanlewis** [reply]
I understand what's happening now: I have top-level `additionalBranchPrefix` and `commitMessagePrefix` including `{{parentDir}}` which does what I want: one branch per component. So I assume what I need to do is override that for the `lockFileMaintenance` case...

**Reply by rarkins** [reply, marked as answer]
Yes, try a package rule with matchUpdateTypes=lockFileMaintenance and additionalBranchPrefix=null

**Reply by allanlewis** [reply]
Thanks, I got this working using a package rule and some changes to `additionalBranchPrefix` and `commitMessagePrefix`.
